core/src/hlist.rs

```diff
@@ -643,7 +643,43 @@ where
     }
 }
 
-/// Trail that allow for mapping over a data structure using mapping functions stored in another
+// Thin wrapper for discriminating against just F
+pub struct Poly<T>(pub T);
+
+// Special index type for Poly; essentially ignored
+pub enum PolyIndex {}
```

Comment:
Um... hm... now that I think about it, why does HMappable even take an index?

Reply:
I can't remember now, but I think I needed to use it for disambiguation of `impl`s, but I don't remember with what now.

Reply:
I'm looking further into it right now but it seems to me right now it was just something funny about how the `for &HCons/&HNil` impls were written, which can be easily corrected.

(though this is out of scope for this PR; I'll make an issue or PR)